pandas/core/arrays/integer.py

```diff
@@ -742,17 +742,28 @@ def integer_arithmetic_method(self, other):
                     raise TypeError("can only perform ops with numeric values")
 
             # nans propagate
-            if mask is None:
+            if omask is None:
                 mask = self._mask.copy()
             else:
-                mask = self._mask | mask
+                mask = self._mask | omask
 
-            # 1 ** np.nan is 1. So we have to unmask those.
             if op_name == "pow":
-                mask = np.where(self == 1, False, mask)
+                # 1 ** x is 1.
+                mask = np.where((self._data == 1) & ~self._mask, False, mask)
+                # x ** 0 is 1.
+                if omask is not None:
+                    mask = np.where((other == 0) & ~omask, False, mask)
+                else:
+                    mask = np.where(other == 0, False, mask)
 
             elif op_name == "rpow":
-                mask = np.where(other == 1, False, mask)
+                # 1 ** x is 1.
+                if omask is not None:
+                    mask = np.where((other == 1) & ~omask, False, mask)
```

Comment:
shouldn't this check for `self._data == 1` ?

Reply:
Ah, no, because it is for the reversed op? (maybe add a comment)

Reply:
Your call on whether a comment is needed. It's clear IMO, though I wrote it :)